### PR TITLE
Add API docs to sidebar

### PR DESCRIPTION
**Pull request by spacey-sooty**

### CODE REVIEW

**Comment by spacey-sooty** [comment]
image
hmmmm

**Comment by mcm001** [comment]
```
/home/runner/work/photonvision/photonvision/docs/source/index.rst:114: WARNING: toctree contains reference to nonexisting document ' :maxdepth: 1'
/home/runner/work/photonvision/photonvision/docs/source/index.rst:114: WARNING: toctree contains reference to nonexisting document ' :caption: API Documentation'
/home/runner/work/photonvision/photonvision/docs/source/index.rst:114: WARNING: toctree contains reference to nonexisting document ' :hidden:'
```

**Comment by mcm001** [comment]
What's odd is that wpilib docs do it the same way here https://raw.githubusercontent.com/wpilibsuite/frc-docs/main/source/index.rst

```suggestion
.. toctree::
   :maxdepth: 1
   :caption: API Docs
   :hidden:

   WPILib Java API Docs <https://github.wpilib.org/allwpilib/docs/release/java/index.html>
   WPILib C++ API Docs <https://github.wpilib.org/allwpilib/docs/release/cpp/index.html>
   WPILib Python API Docs <https://robotpy.readthedocs.io/projects/robotpy/en/stable/>
```

**Comment by spacey-sooty** [comment]
Ngl I got no clue 😭
The magic that is sphinx and rst is beyond me

**Comment by GrahamSH-LLK** [comment]
> What's odd is that wpilib docs do it the same way here https://raw.githubusercontent.com/wpilibsuite/frc-docs/main/source/index.rst
> 
> ```
> .. toctree::
>    :maxdepth: 1
>    :caption: API Docs
>    :hidden:
> 
>    WPILib Java API Docs <https://github.wpilib.org/allwpilib/docs/release/java/index.html>
>    WPILib C++ API Docs <https://github.wpilib.org/allwpilib/docs/release/cpp/index.html>
>    WPILib Python API Docs <https://robotpy.readthedocs.io/projects/robotpy/en/stable/>
> ```

Ours is misaligned (looks like one space off)

File: docs/source/index.rst [hunk 1]
```diff
@@ -110,3 +110,12 @@ PhotonVision is licensed under the `GNU GPL v3 <https://www.gnu.org/licenses/gpl
    docs/additional-resources/config
    docs/additional-resources/nt-api
    docs/contributing/index
+
+.. toctree::
+  :maxdepth: 1
+  :caption: API Documentation
+  :hidden:
```
Comment (spacey-sooty) [inline docs/source/index.rst:117]:
```suggestion
   :maxdepth: 1
   :caption: API Documentation
   :hidden:
```

File: docs/source/index.rst [hunk 2]
```diff
@@ -110,3 +110,12 @@ PhotonVision is licensed under the `GNU GPL v3 <https://www.gnu.org/licenses/gpl
    docs/additional-resources/config
    docs/additional-resources/nt-api
    docs/contributing/index
+
+.. toctree::
+   :maxdepth: 1
+   :caption: API Documentation
+   :hidden:
```
Comment (spacey-sooty) [inline docs/source/index.rst:117]:
```suggestion
 :maxdepth: 1
 :caption: API Documentation
 :hidden:
```

**Comment by spacey-sooty** [comment]
PROGRESSS

File: docs/source/index.rst [hunk 3]
```diff
@@ -110,3 +110,12 @@ PhotonVision is licensed under the `GNU GPL v3 <https://www.gnu.org/licenses/gpl
...
    docs/contributing/index
+
+.. toctree::
+ :maxdepth: 1
+ :caption: API Documentation
+ :hidden:
+
+ `Java <https://javadocs.photonvision.org>`_
+
+ `C++ <https://cppdocs.photonvision.org/>`_
```
Comment (spacey-sooty) [inline docs/source/index.rst:121]:
```suggestion
  C++ <https://cppdocs.photonvision.org/>
```

File: docs/source/index.rst [hunk 4]
```diff
@@ -110,3 +110,12 @@ PhotonVision is licensed under the `GNU GPL v3 <https://www.gnu.org/licenses/gpl
    docs/additional-resources/config
    docs/additional-resources/nt-api
    docs/contributing/index
+
+.. toctree::
+ :maxdepth: 1
+ :caption: API Documentation
+ :hidden:
+
+ `Java <https://javadocs.photonvision.org>`_
```
Comment (spacey-sooty) [inline docs/source/index.rst:119]:
```suggestion
  Java <https://javadocs.photonvision.org>
```

**Comment by spacey-sooty** [comment]
How do I undraft a pr on mobile...
This is ready for review

**Review by mcm001 — APPROVED** [review]

**Pull request merged**